src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -179,17 +164,49 @@ private ProjectInstance TryGetProjectInstance(string projectPath, Dictionary<str
             return null;
         }
 
+        /// <returns>Returns true if the path exists and is a project file type.</returns> 
         private bool IsValidProjectFilePath(string path)
         {
             return File.Exists(path) && Path.GetExtension(path).EndsWith("proj");
         }
 
+        /// <param name="targetProject">The project which we want to get the potential modern (NET 5.0/Core+) TFM of.</param>
+        /// <returns>
+        /// Returns the target framework of the project.
+        /// This will return 0 if there are no TFM of the form netX.0.
+        /// If the project is multi-targeted, it will return the maximum target framework number (e.g. 8 for net8.0) of all modern TFMS in the project.
+        /// </returns>
+        /// <remarks>If the naming schema of TFM changes, this will not work. Yeah, it's not great.</remarks>
+        private float GetProjectMaxModernTargetFramework(ProjectInstance targetProject)
+        {
+            string multiTargetingFrameworks = targetProject.GetPropertyValue(MSBuildPropertyNames.TARGET_FRAMEWORKS);
+            string targetFramework = targetProject.GetPropertyValue(MSBuildPropertyNames.TARGET_FRAMEWORK);
+            if (!string.IsNullOrEmpty(multiTargetingFrameworks))
```

Comment:
One tested concern: 
If you have 8.0 as your publish TFM but it's not in your `TargetFrameworks` then it will fail anyways, so this behaves as expected (will use Release for 8.0 if you specify both `TargetFrameworks` and `TargetFramework`)